resolve/matrix/MatrixHandler.cpp

```diff
@@ -95,6 +95,12 @@ namespace ReSolve {
     isHipEnabled_ = true;
   }
 #endif
+  /**
+   * @brief Set the flag indicating that the matrix values have changed.
+   *
+   * @param[in] isValuesChanged - true if the values have changed, false otherwise
+   * @param[in] memspace - Device where the flag is set
```

Comment:
I think we need to document this with a little more detail. Try something like this:
```suggestion
   * @brief Set the flag indicating that the matrix values have changed.
   *
   * If set to true, next invocation of `matvec` method will trigger re-allocation of the
   * matrix descriptor object. Use if the matrix changes or if matrix object internal pointers
   * to the raw matrix data change. This method has no effect if you are using CPU backend. 
   *
   * @warning This is expert-level method. Use only if you know what you are doing.
   *
   * @param[in] isValuesChanged - true if the values have changed, false otherwise
   * @param[in] memspace - Memory where values have changed
```
@kswirydo, please chime in.